templates/typescript/webcrypto.ts

```diff
@@ -0,0 +1,66 @@
+let subtle = self.crypto.subtle;
+
+async function handleRequest() {
+  let msg = "alice and bob"
+  let hmacresult = await generateSignandVerify({
+    name: "HMAC",
+    hash: "sha-256"
+  })
+  console.log("Result of HMAC generate, sign, verify: ", hmacresult)
+  let aesresult = await generateEncryptDecrypt({
+    name: "AES-GCM",
+    length: 256
+  }, msg)
+  var dec = new TextDecoder();
+  if (msg == dec.decode(aesresult)) {
+    console.log("AES encrypt decrypt successful")
+  } else {
+    console.log("AES encrypt decrypt failed")
+  }
+  return new Response()
+}
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest())
+})
+
+async function generateSignandVerify(algorithm: any) {
+  let rawMessage = "alice and bob"
+  let key = await subtle.generateKey(algorithm, true, ["sign", "verify"]);
```

Comment:
could we cast this `as CryptoKey` here so we don't have to do it in the `subtle` calls?